docs/ugcstor-csi.md

```diff
@@ -0,0 +1,487 @@
+<ol>
+ <li>Priority class needs to be created beforehand. In this case, high-priority and ultra-priority priority classes should exist.</li>
+ <li>The index starts from 0 for @.spec.pools list.</li>
+</ol>
+
+```
+apiVersion: cstor.openebs.io/v1
+kind: CStorPoolCluster
+metadata:
+  name: demo-pool-cluster
+  namespace: openebs
+spec:
+
+  priorityClassName: high-priority 
+
+  pools:
+    - nodeSelector:
+        kubernetes.io/hostname: worker-node-1
+
+      dataRaidGroups:
+      - cspiBlockDevices:
+          - blockDeviceName: blockdevice-ada8ef910929513c1ad650c08fbe3f36
+          - blockDeviceName: blockdevice-ada8ef910929513c1ad650c08fbe3f37
+
+      poolConfig:
+        dataRaidGroupType: mirror
+
+    - nodeSelector:
+        kubernetes.io/hostname: worker-node-2
+
+      dataRaidGroups:
+      - cspiBlockDevices:
+          - blockDeviceName: blockdevice-ada8ef910929513c1ad650c08fbe3f39
+          - blockDeviceName: blockdevice-ada8ef910929513c1ad650c08fbe3f40
+
+      poolConfig:
+        dataRaidGroupType: mirror
+
+    - nodeSelector:
+        kubernetes.io/hostname: worker-node-3
+
+      dataRaidGroups:
+      - cspiBlockDevices:
+          - blockDeviceName: blockdevice-ada8ef910929513c1ad650c08fbe3f42
+          - blockDeviceName: blockdevice-ada8ef910929513c1ad650c08fbe3f43
+
+      poolConfig:
+        dataRaidGroupType: mirror
+        priorityClassName: utlra-priority
+```
+<font size="4">**Example configuration for Compression:**</font>
+Compression values can be set at pool level only. There is no override mechanism like it was there in case of tolerations, resources, auxResources and priorityClass. Compression value must be one of on,off,lzjb,gzip,gzip-[1-9],zle and lz4.
+
+**Note:** lz4 is the default compression algorithm that is used if the compression field is left unspecified on the cspc. Below is the sample yaml which has compression specified.
+
+```
+
+apiVersion: cstor.openebs.io/v1
+kind: CStorPoolCluster
+metadata:
+  name: demo-pool-cluster
+  namespace: openebs
+spec:
+  pools:
+    - nodeSelector:
+        kubernetes.io/hostname: worker-node-1
+
+      dataRaidGroups:
+      - cspiBlockDevices:
+          - blockDeviceName: blockdevice-ada8ef910929513c1ad650c08fbe3f36
+          - blockDeviceName: blockdevice-ada8ef910929513c1ad650c08fbe3f37
+
+      poolConfig:
+        dataRaidGroupType: mirror
+        compression: lz
+```
+<font size="4">**Example configuration for Read Only Threshold:**</font>
+RO threshold can be set in a similar manner like compression. ROThresholdLimit is the threshold(percentage base) limit for pool read only mode. If ROThresholdLimit(%) amount of pool storage is consumed then the pool will be set to readonly. If ROThresholdLimit is set to 100 then entire pool storage will be used. By default it will be set to 85% i.e when unspecified on the CSPC.ROThresholdLimit value will be 0 < ROThresholdLimit <= 100. Following CSPC yaml has the ReadOnly Threshold percentage specified.
```

Comment:
How about below one?
```suggestion
RO threshold can be set in a similar manner like compression. ROThresholdLimit is the threshold(percentage base) limit for pool read only mode. If storage consumption is greater than ROThresholdLimit(%) then the pool will be marked as readonly. If ROThresholdLimit is set to 100 then entire pool storage will be used. By default, it will be set to 85% i.e when unspecified on the CSPC.ROThresholdLimit value will be 0 < ROThresholdLimit <= 100. Following CSPC yaml has the ReadOnly Threshold percentage specified.
```